nx/lib/nx/defn/evaluator.ex

```diff
@@ -42,12 +42,18 @@ defmodule Nx.Defn.Evaluator do
   def __compile__(_key, vars, fun, opts) do
```

Comment:
self note: if we're already in Elixir 1.13+ I believe we can use Keyword.validate here